ydb/core/grpc_services/grpc_request_check_actor.h

```diff
@@ -545,30 +546,22 @@ class TGrpcRequestCheckActor
                         << ", database: " << CheckedDatabaseName_
                         << ", user: " << TBase::GetUserSID()
                         << ", from ip: " << GrpcRequestBaseCtx_->GetPeerName());
-            return {false, std::nullopt};
+            return false;
         }
 
         const ui32 access = NACLib::ConnectDatabase;
         const auto& parsedToken = TBase::GetParsedToken();
         if (parsedToken && SecurityObject_->CheckAccess(access, *parsedToken)) {
-            return {false, std::nullopt};
+            return false;
         }
 
-        const TString error = TStringBuilder()
-            << "User has no permission to perform query on this database"
-            << ", database: " << CheckedDatabaseName_
-            << ", user: " << TBase::GetUserSID()
-            << ", from ip: " << GrpcRequestBaseCtx_->GetPeerName();
-        LOG_INFO(*TlsActivationContext, NKikimrServices::GRPC_PROXY_NO_CONNECT_ACCESS, "%s", error.c_str());
-
         Counters_->IncDatabaseAccessDenyCounter();
 
         if (!AppData()->FeatureFlags.GetCheckDatabaseAccessPermission()) {
-            return {false, std::nullopt};
+            return false;
         }
 
-        LOG_INFO(*TlsActivationContext, NKikimrServices::GRPC_SERVER, "%s", error.c_str());
-        return {true, MakeIssue(NKikimrIssues::TIssuesIds::ACCESS_DENIED, error)};
+        return true;
```

Comment:
There we deny user the ability to differentiate between two causes of ACCESS_DENIED:
1. inability to work with the database as a whole (no connect-db right)
2. no sufficient rights to work with the specific path in the database (all other schema rights)

I think that ACCESS_DENIED for no connect-db right should come with _some_ message specific to that cause. Empty ACCESS_DENIED is not descriptive enough.